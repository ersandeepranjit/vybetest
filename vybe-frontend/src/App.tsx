
import './App.css'
import Chart from "react-apexcharts";
import {useApp} from './useApp';



function App() {
  const {
    marketCapChart,
    tpsChart,
    balanceChart,
  } = useApp();
  return (
    <>
      <div>
      {marketCapChart &&
      <>
      <h1>Market Capital (USD)</h1>
      <div className="card">
        <Chart
              options={marketCapChart?.options}
              series={marketCapChart?.series}
              type="pie"
              width="500"
            />
       
        </div>
        </>
         }
      </div>
      { tpsChart &&
      <>
      <h1>Transaction Per second</h1>
      <div className="card">
         <Chart
          options={tpsChart?.options}
          series={tpsChart?.series}
          type="line"
        />
        </div>
        </>
        }
        {balanceChart&& 
       <>
       <h1>Wallet Balance</h1>
       <div className="card"> 
       <Chart
          options={balanceChart.options}
          series={balanceChart.series}
          type="bar"
          width="1000"
       />
        </div>
        </>
        } 
      
    </>
  )
}

export default App
