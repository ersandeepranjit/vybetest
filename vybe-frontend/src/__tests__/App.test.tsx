import { render, screen } from "@testing-library/react";
import App from "../App";
import  React from "react"
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';
import ResizeObserver from 'resize-observer-polyfill';

import ReactApexChart from "react-apexcharts";
import ApexCharts from "apexcharts";

global.ResizeObserver = ResizeObserver;
global.React = React;

jest.mock('react-apexcharts', () => jest.fn(() => { return null }));
jest.mock('apexcharts', () => ({ exec: jest.fn(() => { return new Promise((resolve, reject) => { resolve("uri") }) }) }));
const mock = {
balanceChart: {
	series: [{
		data: [400, 430]
	  }],
	options: {
		chart: {
		  type: 'bar',
		  height: 350
		},
		plotOptions: {
		  bar: {
			borderRadius: 4,
			horizontal: true,
		  }
		},
		dataLabels: {
		  enabled: false
		},
		xaxis: {
		  categories: [ "7VHUFJHWu2CuExkJcJrzhQPJ2oygupTWkL2A2For4BmE",
		  "9WzDXwBbmkg8ZTbNMqUxvQRAyrZzDsGYdLVL9zYtAWWM"
		  ],
		}
	  }
}, 
marketCapChart: {
	series: [20059256.588990252, 1057680.4350760388],
    options: {
              chart: {
                width: 380,
                type: 'pie',
              },
              labels: ["7xKXtg2CW87d97TXJSDpbD5jBkheTqA83TZRuJosgAsU","CWE8jPTUYhdCTZYWPTe1o5DFqfdjzWKc9WKz6rSjQUdG"],
              responsive: [{
                breakpoint: 480,
                options: {
                  chart: {
                    width: 200
                  },
                  legend: {
                    position: 'bottom'
                  }
                }
              }]
			}
},
tpsChart: {
          
	series: [{
		name: "TPS",
		data: [43341,3939]
	}],
	options: {
	  chart: {
		height: 350,
		type: 'line',
		zoom: {
		  enabled: false
		}
	  },
	  dataLabels: {
		enabled: false
	  },
	  stroke: {
		curve: 'straight'
	  },
	  title: {
		text: 'Transaction per Second',
		align: 'left'
	  },
	  grid: {
		row: {
		  colors: ['#f3f3f3', 'transparent'], // takes an array which will be repeated on columns
		  opacity: 0.5
		},
	  },
	  xaxis: {
		categories: [199344569,199344446],
	  }
	}
 }
}

jest.mock('../useApp', () => ({
		useApp:()=>{
			return mock;
		}
}));

const useStateSpy = jest.spyOn(React, "useState");

describe("App", () => {
  
      
  it("should work as expected", () => {
    shallow(<App />);
  });
  
  it("should render the Market Cap chart",()=>{
		const wrapper = shallow(<App/>);
		const  marketCap = <h1>Market Capital</h1>
		expect(wrapper.contains(marketCap)).toEqual(true);
  });
  it("should render the TPS chart",()=>{
	const wrapper = shallow(<App/>);
	const  tps = <h1>Transaction Per second</h1>
	expect(wrapper.contains(tps)).toEqual(true);
});
it("should render the Balance chart",()=>{
	const wrapper = shallow(<App/>);
	const  balance = <h1>Wallet Balance</h1>
	expect(wrapper.contains(balance)).toEqual(true);
});
});

describe("Snapshots",()=>{
it("should renders correctly",()=>{
	const tree =shallow(<App/>);
	expect(toJSON(tree)).toMatchSnapshot();
	});
});