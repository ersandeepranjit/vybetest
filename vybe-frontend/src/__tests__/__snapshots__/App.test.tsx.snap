// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots should renders correctly 1`] = `
<Fragment>
  <div>
    <h1>
      Market Capital (USD)
    </h1>
    <div
      className="card"
    >
      <mockConstructor
        options={
          Object {
            "chart": Object {
              "type": "pie",
              "width": 380,
            },
            "labels": Array [
              "7xKXtg2CW87d97TXJSDpbD5jBkheTqA83TZRuJosgAsU",
              "CWE8jPTUYhdCTZYWPTe1o5DFqfdjzWKc9WKz6rSjQUdG",
            ],
            "responsive": Array [
              Object {
                "breakpoint": 480,
                "options": Object {
                  "chart": Object {
                    "width": 200,
                  },
                  "legend": Object {
                    "position": "bottom",
                  },
                },
              },
            ],
          }
        }
        series={
          Array [
            20059256.58,
            1057680.43,
          ]
        }
        type="pie"
        width="500"
      />
    </div>
  </div>
  <h1>
    Transaction Per second
  </h1>
  <div
    className="card"
  >
    <mockConstructor
      options={
        Object {
          "chart": Object {
            "height": 350,
            "type": "line",
            "zoom": Object {
              "enabled": false,
            },
          },
          "dataLabels": Object {
            "enabled": false,
          },
          "grid": Object {
            "row": Object {
              "colors": Array [
                "#f3f3f3",
                "transparent",
              ],
              "opacity": 0.5,
            },
          },
          "stroke": Object {
            "curve": "straight",
          },
          "title": Object {
            "align": "left",
            "text": "Transaction per Second",
          },
          "xaxis": Object {
            "categories": Array [
              199344569,
              199344446,
            ],
          },
        }
      }
      series={
        Array [
          Object {
            "data": Array [
              43341,
              3939,
            ],
            "name": "TPS",
          },
        ]
      }
      type="line"
    />
  </div>
  <h1>
    Wallet Balance
  </h1>
  <div
    className="card"
  >
    <mockConstructor
      options={
        Object {
          "chart": Object {
            "height": 350,
            "type": "bar",
          },
          "dataLabels": Object {
            "enabled": false,
          },
          "plotOptions": Object {
            "bar": Object {
              "borderRadius": 4,
              "horizontal": true,
            },
          },
          "xaxis": Object {
            "categories": Array [
              "7VHUFJHWu2CuExkJcJrzhQPJ2oygupTWkL2A2For4BmE",
              "9WzDXwBbmkg8ZTbNMqUxvQRAyrZzDsGYdLVL9zYtAWWM",
            ],
          },
        }
      }
      series={
        Array [
          Object {
            "data": Array [
              400,
              430,
            ],
          },
        ]
      }
      type="bar"
      width="1000"
    />
  </div>
</Fragment>
`;
